docker/package/tezos_baking_wizard.py

```diff
@@ -513,38 +560,79 @@ def bootstrap_node(self):
+    # if it can be overwritten.
+    def check_baker_account(self):
+        address = subprocess.run(
+            shlex.split("sudo -u tezos tezos-client show address baker --show-secret"),
+            capture_output=True,
+        )
+        if address.returncode == 0:
+            value_regex = b"(?:" + ledger_regex + b")|(?:" + secret_key_regex + b")"
+            value = re.search(value_regex, address.stdout).group(0)
+            address = re.search(address_regex, address.stdout).group(0)
+            print()
+            print("An account with the 'baker' alias already exists.")
+            print("Its current value is", value.decode("utf-8"))
+            print("With a corresponding address:", address.decode("utf-8"))
+
+            return yes_or_no("Should it be replaced with the new import? <y/N> ", "no")
+
     # Importing the baker key
     def import_baker_key(self):
-        if self.config["key_import_mode"] == "secret-key":
-            self.check_baker_account(
-                "sudo -u tezos tezos-client import secret key baker "
-                + self.config["secret_key"]
-            )
-        elif self.config["key_import_mode"] == "json":
-            self.check_baker_account(
-                "sudo -u tezos tezos-client activate account baker with "
-                + self.config["json_filepath"]
-            )
 
-        else:
-            print()
-            input("Please open the Tezos Baking app on your ledger, then hit Enter.")
-            self.check_baker_account(
-                "sudo -u tezos tezos-client import secret key baker "
-                + self.config["ledger_url"]
-            )
-            proc_call("sudo -u tezos tezos-client setup ledger to bake for baker")
+        replace_baker_key = self.check_baker_account()
+        valid_choice = False
+
+        while replace_baker_key and not valid_choice:
+
+            self.query_step(key_mode_query)
+
+            if self.config["key_import_mode"] == "secret-key":
+                self.query_step(secret_key_query)
+                proc_call(
+                    "sudo -u tezos tezos-client import secret key baker "
+                    + self.config["secret_key"]
+                    + " --force"
+                )
+            elif self.config["key_import_mode"] == "json":
+                if self.config["network"] == "mainnet":
+                    print()
+                    print(
+                        "Baking on the mainnet with a JSON file from the testnet faucet is impossible."
+                    )
+                    print("Please choose one of the other options.")
+                    print()
+                    continue
+
+                self.query_step(json_filepath_query)
+                proc_call(
+                    "sudo -u tezos tezos-client activate account baker with "
+                    + self.config["json_filepath"]
+                    + " --force"
+                )
```

Comment:
I purposefully give the wizard a wrong path (to a file that exists) and this results in a failure of the whole wizard.

Can't we instead make sure that if `proc_call` like this fails we repeat the choice (I guess in this case the whole `import_baker_key`)?

More in general, I think this should be done for any call we do to `tezos-client` as there are several things that could go wrong and I think it's always better to re-give the choice instead of failing the whole process.

Reply:
I did what you suggested for importing the key, but what do you suggest to do about other `tezos-client` calls? E.g. one such call is made in `register_baker`, and if that fails after importing the key, that's (most likely) not the user's fault. What choice should be re-given then?

Reply:
mmmh good point, in general I would think that the best UX would be given by providing an option to either try again or go to the previous step (importing the baker key again).

However we need to think of each individual case IMO, in the `register_baker` example maybe we should just go back to the import step directly? 
Simply because that will recognize that there is already a `baker` and give the option to skip the import (so jumping to retying the registering)